python_modules/dagster/dagster/_core/definitions/asset_layer.py

```diff
@@ -77,25 +77,26 @@ def from_graph_and_assets_node_mapping(
 
             for output_name, asset_key in assets_def.node_keys_by_output_name.items():
                 # resolve graph output to the op output it comes from
-                inner_output_def, inner_node_handle = (
-                    computation.full_node_def.resolve_output_to_origin(
-                        output_name, handle=node_handle
-                    )
+                full_node_def = computation.full_node_def
+                inner_output_def, inner_node_handle = full_node_def.resolve_output_to_origin(
+                    output_name, handle=node_handle
                 )
                 node_output_handle = NodeOutputHandle(
                     node_handle=inner_node_handle, output_name=inner_output_def.name
                 )
 
                 asset_keys_by_node_output_handle[node_output_handle] = asset_key
 
-                asset_key_by_input.update(
-                    {
-                        input_handle: asset_key
-                        for input_handle in computation.full_node_def.resolve_output_to_destinations(
-                            output_name, node_handle
-                        )
-                    }
+                destinations = computation.full_node_def.resolve_output_to_destinations(
+                    output_name, node_handle
                 )
+                for destination in destinations:
+                    asset_key_by_input[destination] = asset_key
+                    if isinstance(full_node_def, GraphDefinition):
+                        for op_input_handle in full_node_def.get_node(
+                            destination.node_handle.pop()
+                        ).definition.resolve_input_to_destinations(destination):
+                            asset_key_by_input[op_input_handle] = asset_key
```

Comment:
this is a behavior change right? does this fix a bug? Worth adding/updating a test if so

Reply:
It shouldn't be a behavior change. It just means that our internal derived dict that tracks the relationship between node inputs and asset keys now has some additional keys for leaf node inputs.